frame/plasm-staking/src/lib.rs

```diff
@@ -570,13 +624,195 @@ impl<T: Trait> Module<T> {
+        let mut total_imbalance = <PositiveImbalanceOf<T>>::zero();
+        let operators_reward =
+            Perbill::from_rational_approximation(BalanceOf::<T>::from(4), BalanceOf::<T>::from(5))
+                * total_payout;
+        let nominators_reward = total_payout
+            .checked_sub(&operators_reward)
+            .unwrap_or(BalanceOf::<T>::zero());
+        let staked_contracts = <StakedContracts<T>>::enumerate()
+            .collect::<Vec<(T::AccountId, Exposure<T::AccountId, BalanceOf<T>>)>>();
+        let total_staked = staked_contracts
+            .iter()
+            .fold(BalanceOf::<T>::zero(), |sm, (_, e)| {
+                sm.checked_add(&e.total).unwrap_or(sm)
+            });
+
+        for (c, e) in staked_contracts.iter() {
+            let reward =
+                Perbill::from_rational_approximation(e.total, total_staked) * operators_reward;
+            total_imbalance.subsume(Self::reward_contract(&c, reward));
+        }
+
+        let nominate_totals = staked_contracts.iter().fold(
+            BTreeMap::<T::AccountId, BalanceOf<T>>::new(),
+            |mp, (_, e)| {
+                e.others.iter().fold(mp, |mut m, ind| {
+                    if m.contains_key(&ind.who) {
+                        if let Some(m) = m.get_mut(&ind.who) {
+                            *m += ind.value;
+                        }
+                    } else {
+                        m.insert(ind.who.clone(), ind.value);
+                    }
+                    return m;
+                })
+            },
+        );
+
+        for (n, t) in nominate_totals.iter() {
+            let reward = Perbill::from_rational_approximation(*t, total_staked) * nominators_reward;
+            total_imbalance
+                .subsume(Self::make_payout(n, reward).unwrap_or(PositiveImbalanceOf::<T>::zero()));
+        }
+        let total_payout = total_imbalance.peek();
+
+        let rest = max_payout.saturating_sub(total_payout);
+
+        T::Reward::on_unbalanced(total_imbalance);
+        T::RewardRemainder::on_unbalanced(T::Currency::issue(rest));
+    }
+
+    fn elected_operators() {
+        let nominations = <DappsNominations<T>>::enumerate()
+            .filter(|(_, n)| !n.suppressed)
+            .collect::<Vec<(T::AccountId, Nominations<T::AccountId>)>>();
+        let nominators = nominations
+            .iter()
+            .cloned()
+            .map(|(s, _)| s)
+            .collect::<Vec<T::AccountId>>();
+        let nominators_to_staking = nominators
+            .into_iter()
+            .map(|n| {
+                if let Some(ctrl) = Self::bonded(&n) {
+                    if let Some(ledger) = Self::ledger(&ctrl) {
+                        return (n, ledger.active);
+                    }
+                }
+                (n, BalanceOf::<T>::zero())
+            })
+            .collect::<BTreeMap<T::AccountId, BalanceOf<T>>>();
+
+        let staked_contracts = nominations.iter().fold(
```

Comment:
This construction a bit difficult to read, what’s about to introduce additional function with readable names or don’t use one char variables

Reply:
Thanks! I refactored this function.